pf/tfbridge/ignore_changes.go

```diff
@@ -0,0 +1,301 @@
+		if ic.isIgnoredBy(p, ap) {
+			return true
+		}
+	}
+	return false
+}
+
+func (ic *ignoreChanges) isIgnoredBy(pattern resource.PropertyPath, ap *tftypes.AttributePath) bool {
+	mc := &matchingContext{
+		remainingPattern: pattern,
+		schemaStepper:    ic.ss,
+	}
+	match, remain, err := tftypes.WalkAttributePath(mc, ap)
+	if err != nil {
+		return false
+	}
+	if remain != nil && len(remain.Steps()) > 0 {
+		return false
+	}
+	if m, ok := match.(*matchingContext); ok {
+		if !m.matchFailed {
+			return true
+		}
+	}
+	return false
+}
+
+type schemaStepper interface {
+	Property(resource.PropertyKey) schemaStepper
+	Element() schemaStepper
+	PropertyKey(convert.TerraformPropertyName) resource.PropertyKey
+}
+
+type typeSchemaStepper struct {
+	renames convert.PropertyNames
+	schema  *schema.PackageSpec
+	t       *schema.TypeSpec
+}
+
+var _ schemaStepper = (*typeSchemaStepper)(nil)
+
+func (p *typeSchemaStepper) Element() schemaStepper {
+	if p.t.AdditionalProperties != nil {
+		return typeStepper(p.renames, p.schema, p.t.AdditionalProperties)
+	}
+	if p.t.Items != nil {
+		return typeStepper(p.renames, p.schema, p.t.Items)
+	}
+	return nil
+}
+
+func (*typeSchemaStepper) Property(k resource.PropertyKey) schemaStepper {
+	// there are no named properties here, those are supported by namedEntitySchemaStepper
+	return nil
+}
+
+func (p *typeSchemaStepper) PropertyKey(n convert.TerraformPropertyName) resource.PropertyKey {
+	// translate verbatim, no tables
+	return resource.PropertyKey(tokens.Name(n))
+}
+
+// Matching names at a Resource or named object type.
+type namedEntitySchemaStepper struct {
+	schema  *schema.PackageSpec
+	token   tokens.Token
+	renames convert.PropertyNames
+}
+
+func (r *namedEntitySchemaStepper) Property(k resource.PropertyKey) schemaStepper {
+	// If this is a Resource..
+	if rr, ok := r.schema.Resources[string(r.token)]; ok {
+		// Only InputProperties here, not Properties because the docs state that:
+		//
+		//     The ignoreChanges option only applies to resource inputs, not outputs.
+		//
+		// See https://www.pulumi.com/docs/intro/concepts/resources/options/ignorechanges/
+		if p, ok := rr.InputProperties[string(k)]; ok {
+			return typeStepper(r.renames, r.schema, &p.TypeSpec)
+		}
+	}
+	// If this is a named type..
+	if tt, ok := r.schema.Types[string(r.token)]; ok {
+		if p, ok := tt.Properties[string(k)]; ok {
+			return typeStepper(r.renames, r.schema, &p.TypeSpec)
+		}
+	}
+	return nil
+}
+
+func (*namedEntitySchemaStepper) Element() schemaStepper {
+	// list or map element path does not make sense for resources, data sources, named object types
+	return nil
+}
+
+func (r *namedEntitySchemaStepper) PropertyKey(n convert.TerraformPropertyName) resource.PropertyKey {
+	return r.renames.PropertyKey(r.token, n, nil)
+}
+
+func typeStepper(renames convert.PropertyNames, s *schema.PackageSpec, t *schema.TypeSpec) schemaStepper {
+	// properties may refer to named object types via refs
+	if t.Ref != "" {
```

Comment:
`t.Ref` can refer to `"#/resources/"` as well. I think we can handle that easily in this first draft.

Reply:
Hmm that's interesting. Do you have an example where we use that? I can search some schemas.

Reply:
We use it in `awsx` a lot. I'm not sure if this ever appears in a TF bridged provider. We could just provided a custom error message telling the user to let us know if they see it. After seeing that TF bridged providers never use this functionality, it doesn't seem like a high priority.

Reply:
Yeah perfect I think just a solid comment could work here for now.